dwave/optimization/src/nodes/creation.cpp

```diff
@@ -217,12 +222,12 @@ std::pair<double, double> ARangeNode::minmax(
         // The direction that we're stepping determines the min/max
         if (step_low > 0) {
             const double low = start_low;
-            const double high = stop_high;
+            const double high = stop_high - step_low;
```

Comment:
Why did you add the `- step_low` to this? If I have
```
start_low, start_high = 0, 0
stop_low, stop_high = 5, 5
step_low, step_high = 2, 4
```
Then the ARangeNode `high` should be 4, right? But this would set it to `stop_high - step_low = 3`.

Reply:
You're right. It was an attempt to account for the non-inclusive stop, but it doesn't cover all cases. Will fix.